### PR TITLE
A few more @swift3_migration attribute updates for swift-2.3-branch

### PR DESCRIPTION
**Pull request by benlangmuir**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

A smoke test on macOS does the following:
1. Builds the compiler incrementally.
2. Builds the standard library only for macOS. Simulator standard libraries and
   device standard libraries are not built.
3. lldb is not built.
4. The test and validation-test targets are run only for macOS. The optimized
   version of these tests are not run.

A smoke test on Linux does the following:
1. Builds the compiler incrementally.
2. Builds the standard library incrementally.
3. lldb is built incrementally.
4. The swift test and validation-test targets are run. The optimized version of these
   tests are not run.
5. lldb is tested.

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

rdar://problem/27932410

### CODE REVIEW

**Comment by benlangmuir** [comment]
@swift-ci Please test

**Comment by benlangmuir** [comment]
@moiseev could you please review?

File: stdlib/public/core/CompilerProtocols.swift [hunk 1]
```diff
@@ -21,7 +21,6 @@
 /// Only three types provided by Swift, `Bool`, `DarwinBoolean`, and `ObjCBool`,
 /// conform to `BooleanType`. Expanding this set to include types that
 /// represent more than simple boolean values is discouraged.
-@swift3_migration(renamed="Boolean")
```
Comment (moiseev) [inline stdlib/public/core/CompilerProtocols.swift]: Since the protocol was removed, maybe use the `@swift3_migration(message="")` form?

**Comment by moiseev** [comment]
:shipit:

**Comment by benlangmuir** [comment]
> build-script: error: unrecognized arguments: --skip-build-lldb
@shahmishal did we break testing 2.3 again?

File: stdlib/public/core/CompilerProtocols.swift (hunk 1 above)
Comment (benlangmuir) [inline stdlib/public/core/CompilerProtocols.swift]: Is there anything we can say other than "Removed in Swift 3"?  I didn't see any value in that.

Comment (moiseev) [inline stdlib/public/core/CompilerProtocols.swift]: Right, and we already have an unavailable declaration for the `BoleanType` [here](https://github.com/apple/swift/blob/master/stdlib/public/core/CompilerProtocols.swift#L687).

Comment (benlangmuir) [inline stdlib/public/core/CompilerProtocols.swift]: Okay, I'll leave it as is then.  Lemme know if you think there's anything else I could improve :-)

**Comment by shahmishal** [comment]
@benlangmuir Look into the failure.

```
16:54:49 usage: build-script [-h] [-l] [-b] [-p] [--xctest] [--foundation] [-c]
16:54:49                     [--export-compile-commands] [-d | -r | -R] [--debug-llvm]
16:54:49                     [--debug-swift] [--debug-swift-stdlib] [--debug-lldb]
16:54:49                     [--debug-cmark] [--debug-foundation]
16:54:49                     [--assertions | --no-assertions] [--cmark-assertions]
16:54:49                     [--llvm-assertions] [--no-llvm-assertions]
16:54:49                     [--swift-assertions] [--no-swift-assertions]
16:54:49                     [--swift-stdlib-assertions] [--no-swift-stdlib-assertions]
16:54:49                     [--lldb-assertions] [--no-lldb-assertions] [-x] [-X] [-m]
16:54:49                     [-e] [-t] [-T] [-o] [-S] [-i] [--tvos] [--watchos]
16:54:49                     [--build-subdir PATH] [-j BUILD_JOBS]
16:54:49                     [--darwin-xcrun-toolchain DARWIN_XCRUN_TOOLCHAIN]
16:54:49                     [--cmake CMAKE] [--extra-swift-args EXTRA_SWIFT_ARGS]
16:54:49                     [build_script_impl_args [build_script_impl_args ...]]
16:54:49 
build-script: error: unrecognized arguments: --skip-build-lldb

```

**Comment by shahmishal** [comment]
its due to https://github.com/apple/swift/commit/272587b12ce0be4628b0ec54c989f0bc6eb417ce, its should be fixed now with https://github.com/apple/swift/pull/4425

File: stdlib/public/core/CompilerProtocols.swift (hunk 1 above)
Comment (moiseev) [inline stdlib/public/core/CompilerProtocols.swift]: No. Looks good. I already gave you a :shipit: squirrel. ;-)

**Pull request merged**